PWGLF/Tasks/GlobalEventProperties/heavyionMultiplicity.cxx

```diff
@@ -916,6 +960,125 @@ struct HeavyionMultiplicity {
+      if (!isGenTrackSelected(particle)) {
+        continue;
+      }
+      histos.fill(HIST("hGenMCdndeta"), mcCollision.posZ(), gencent, genoccu, particle.eta(), particle.phi());
+      if (atLeastOne) {
+        histos.fill(HIST("hGenMCAssoRecdndeta"), mcCollision.posZ(), gencent, genoccu, particle.eta(), particle.phi(), static_cast<double>(kGenAll), kNoGenpTVar);
+        if (particle.pt() < KminPtCut) {
+          histos.fill(HIST("hGenMCAssoRecdndeta"), mcCollision.posZ(), gencent, genoccu, particle.eta(), particle.phi(), static_cast<double>(kGenAll), kGenpTup, -10.0 * particle.pt() + 2);
+          histos.fill(HIST("hGenMCAssoRecdndeta"), mcCollision.posZ(), gencent, genoccu, particle.eta(), particle.phi(), static_cast<double>(kGenAll), kGenpTdown, 5.0 * particle.pt() + 0.5);
+        } else {
+          histos.fill(HIST("hGenMCAssoRecdndeta"), mcCollision.posZ(), gencent, genoccu, particle.eta(), particle.phi(), static_cast<double>(kSpAll), kGenpTup);
+          histos.fill(HIST("hGenMCAssoRecdndeta"), mcCollision.posZ(), gencent, genoccu, particle.eta(), particle.phi(), static_cast<double>(kGenAll), kGenpTdown);
+        }
+        int pid = 0;
+        switch (std::abs(particle.pdgCode())) {
+          case PDG_t::kPiPlus:
+            pid = kGenPion;
+            break;
+          case PDG_t::kKPlus:
+            pid = kGenKaon;
+            break;
+          case PDG_t::kProton:
+            pid = kGenProton;
+            break;
+          default:
+            pid = kGenOther;
+            break;
+        }
+        histos.fill(HIST("hGenMCAssoRecdndeta"), mcCollision.posZ(), gencent, genoccu, particle.eta(), particle.phi(), static_cast<double>(pid), kNoGenpTVar);
+      } // Associated with reco col
+    } // track (mcgen) loop
+
+    for (const auto& RecCol : RecCols) {
+      if (!isEventSelected(RecCol)) {
+        continue;
+      }
+      if (RecCol.globalIndex() != mcCollision.bestCollisionIndex()) {
+        continue;
+      }
+      histos.fill(HIST("hRecMCvertexZ"), RecCol.posZ());
+      histos.fill(HIST("hRecMCcentrality"), selColCent(RecCol));
+      histos.fill(HIST("hRecMCvtxzcent"), RecCol.posZ(), selColCent(RecCol), selColOccu(RecCol));
+
+      auto recTracksPart = RecTracks.sliceBy(perCollision, RecCol.globalIndex());
+      std::vector<int> mclabels;
+      for (const auto& Rectrack : recTracksPart) {
+        if (!isTrackSelected(Rectrack)) {
+          continue;
+        }
+        histos.fill(HIST("hRecMCphivseta"), Rectrack.phi(), Rectrack.eta());
+        histos.fill(HIST("hRecMCdndeta"), RecCol.posZ(), selColCent(RecCol), selColOccu(RecCol), Rectrack.eta(), Rectrack.phi(), static_cast<double>(kRecoAll));
+        if (Rectrack.has_mcParticle()) {
+          int pid = 0;
+          auto mcpart = Rectrack.mcParticle();
+          histos.fill(HIST("etaResolution"), Rectrack.eta(), Rectrack.eta() - mcpart.eta());
+          if (mcpart.isPhysicalPrimary()) {
+            pid = kRecoPrimary;
```

Comment:
This value is never used.

Reply:
Hi @abmodak , the value `kRecoPrimary` assigned in the variable `pid` is never used, therefore the assignment is useless.